server/plugin/plugin.go

```diff
@@ -376,14 +379,56 @@ func (p *Plugin) CreateBotDMPost(userID, message, postType string) {
 	}
 }
 
-func (p *Plugin) PostToDo(info *GitHubUserInfo) {
+func (p *Plugin) CheckIfDuplicateDailySummary(userID, text string) (bool, error) {
+	previousSummary, err := p.GetDailySummaryText(userID)
+	if err != nil {
+		return false, err
+	}
+	if previousSummary == text {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (p *Plugin) StoreDailySummaryText(userID, summaryText string) error {
+	if err := p.API.KVSet(userID+dailySummary, []byte(summaryText)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) GetDailySummaryText(userID string) (string, error) {
+	summaryByte, err := p.API.KVGet(userID + dailySummary)
+	if err != nil {
+		return "", err
+	}
+
+	return string(summaryByte), nil
+}
+
+func (p *Plugin) PostToDo(info *GitHubUserInfo, userID string) error {
 	text, err := p.GetToDo(context.Background(), info.GitHubUsername, p.githubConnect(*info.Token))
 	if err != nil {
-		p.API.LogWarn("Failed to get todo text", "userID", info.UserID, "error", err.Error())
-		return
+		return err
 	}
 
+	if info.Settings.DailyReminderOnChange {
+		isSameSummary, err := p.CheckIfDuplicateDailySummary(userID, text)
+		if err != nil {
+			return err
+		}
+		if isSameSummary {
+			return nil
+		}
+		err = p.StoreDailySummaryText(userID, text)
```

Comment:
@hanzei What do you think about storing the daily summary text inside the `GitHubUserInfo` struct, instead of using a separate KV row?

Reply:
Actually that would make it so we are fetching summary from the KV store every time a user wants to do some action with the plugin. Probably fine to keep it as-is